### PR TITLE
Refactored code in preparation for publishing to pypi

### PR DESCRIPTION
**Pull request by zsimic**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,8 +13,5 @@ install:
   - pip install --upgrade h5py
 script:
   - make
-  - export PYTHONPATH=$PWD/python:$PYTHONPATH
-  - python -m doctest python/tools/misc.py
```
Comment (li-zhi) [inline .travis.yml]: We don't need these doctest?

File: README.md [hunk 2]
```diff
@@ -21,12 +22,23 @@ The VDK package has its core feature extraction library written in C, and the re
...
+  - [matplotlib](http://matplotlib.org/1.3.1/index.html) (>=2.0.0)
+  - [pandas](http://pandas.pydata.org/) (>=0.19.2)
+  - [scikit-learn](http://scikit-learn.org/stable/) (>=0.18.1)
+  - [h5py](http://www.h5py.org/) (>=2.6.0)
+
+You will need to install gfortran, freetype, pkg-config, freetype and hdf5 (these can't be compiled from source here):
+- on OSX:
+    - `brew install gcc freetype pkg-config homebrew/science/hdf5` (`gcc` has the `gfortran` compiler needed for scipy)
+    - `echo "backend: TkAgg" > ~/.matplotlib/matplotlibrc` (TODO: how can matplotlib work on OSX without this? see http://matplotlib.org/faq/usage_faq.html#what-is-a-backend)
+- on linux: `sudo apt-get install -y gfortran libfreetype6-dev libhdf5-dev` (TODO: test this)
```
Comment (li-zhi) [inline README.md]: with these, it is guaranteed that we can simply do 'sudo pip install numpy scipy matplotlib pandas'? Is there still a need to ask the user to follow www.scipy.org/install.html?

Comment (zsimic) [inline README.md]: I'm planning on adding/reviewing this part of the doc, I'll add an explanation on how to:
- build completely using a virtual env (no `sudo` needed, but `pip install` into a python virtual env)
- build with `sudo pip install` (typically: Docker)

I'll simplify these instructions as much as possible.

**Comment by li-zhi** [comment]
Awesome!

**Pull request merged**